File: backend/app/schemas/aggregation.py
from datetime import datetime

from pydantic import BaseModel


class AggregationResponse(BaseModel):
    id: str | None
    tenant_id: str
    terminal_id: str | None
    period_type: str
    period_start: datetime
    period_end: datetime
    slot_count: int

    avg_overall: float | None
    avg_sentiment: float | None
    avg_politeness: float | None
    avg_compliance: float | None
    avg_resolution: float | None
    avg_upselling: float | None
    avg_response_time: float | None
    avg_honesty: float | None

    unclear_count: int
    flag_counts: dict

    computed_at: datetime

    model_config = {"from_attributes": True}


class AggregationListResponse(BaseModel):
    items: list[AggregationResponse]
    total: int

File: portal/src/types/aggregation.ts
import { AggregationPeriod } from "./index";

export interface Aggregation {
  id: string;
  tenant_id: string;
  terminal_id: string | null;
  period_type: AggregationPeriod;
  period_start: string;
  period_end: string;
  slot_count: number;
  avg_overall: number | null;
  avg_sentiment: number | null;
  avg_politeness: number | null;
  avg_compliance: number | null;
  avg_resolution: number | null;
  avg_upselling: number | null;
  avg_response_time: number | null;
  avg_honesty: number | null;
  unclear_count: number;
  flag_counts: Record<string, number>;
  computed_at: string;
}

export interface AggregationListParams {
  period_type?: AggregationPeriod;
  period_start?: string;
  period_end?: string;
  terminal_id?: string;
}

export interface AggregationListResponse {
  items: Aggregation[];
  total: number;
}
